src/code/z_map_disp.c

```diff
@@ -1,109 +1,1881 @@
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80102E40.s")
+void MapDisp_DestroyMapI(PlayState* play);
+void MapDisp_InitMapI(PlayState* play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80102E90.s")
+static s32 sLoadTextureBlock_siz[4] = {
+    G_IM_SIZ_4b,
+    G_IM_SIZ_8b,
+    G_IM_SIZ_16b,
+    G_IM_SIZ_32b,
+};
+static s32 sLoadTextureBlock_siz_LOAD_BLOCK[4] = {
+    G_IM_SIZ_4b_LOAD_BLOCK,
+    G_IM_SIZ_8b_LOAD_BLOCK,
+    G_IM_SIZ_16b_LOAD_BLOCK,
+    G_IM_SIZ_32b_LOAD_BLOCK,
+};
+static u32 sLoadTextureBlock_siz_INCR[4] = {
+    G_IM_SIZ_4b_INCR,
+    G_IM_SIZ_8b_INCR,
+    G_IM_SIZ_16b_INCR,
+    G_IM_SIZ_32b_INCR,
+};
+static s32 sLoadTextureBlock_siz_SHIFT[4] = {
+    G_IM_SIZ_4b_SHIFT,
+    G_IM_SIZ_8b_SHIFT,
+    G_IM_SIZ_16b_SHIFT,
+    G_IM_SIZ_32b_SHIFT,
+};
+static u32 sLoadTextureBlock_siz_BYTES[4] = {
+    G_IM_SIZ_4b_BYTES,
+    G_IM_SIZ_8b_BYTES,
+    G_IM_SIZ_16b_BYTES,
+    G_IM_SIZ_32b_BYTES,
+};
+static u32 sLoadTextureBlock_siz_LINE_BYTES[4] = {
+    G_IM_SIZ_4b_LINE_BYTES,
+    G_IM_SIZ_8b_LINE_BYTES,
+    G_IM_SIZ_16b_LINE_BYTES,
+    G_IM_SIZ_32b_LINE_BYTES,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80102EA4.s")
+#define gDPLoadTextureBlock_TEST(pkt, timg, fmt, siz, width, height, pal, cms, cmt, masks, maskt, shifts, shiftt)      \
+    _DW({                                                                                                              \
+        gDPSetTextureImage(pkt, fmt, sLoadTextureBlock_siz_LOAD_BLOCK[siz], 1, timg);                                  \
+        gDPSetTile(pkt, fmt, sLoadTextureBlock_siz_LOAD_BLOCK[siz], 0, 0, G_TX_LOADTILE, 0, cmt, maskt, shiftt, cms,   \
+                   masks, shifts);                                                                                     \
+        gDPLoadSync(pkt);                                                                                              \
+        gDPLoadBlock(pkt, G_TX_LOADTILE, 0, 0,                                                                         \
+                     (((width) * (height) + sLoadTextureBlock_siz_INCR[siz]) >> sLoadTextureBlock_siz_SHIFT[siz]) - 1, \
+                     CALC_DXT(width, sLoadTextureBlock_siz_BYTES[siz]));                                               \
+        gDPPipeSync(pkt);                                                                                              \
+        gDPSetTile(pkt, fmt, sLoadTextureBlock_siz[siz], (((width)*sLoadTextureBlock_siz_LINE_BYTES[siz]) + 7) >> 3,   \
+                   0, G_TX_RENDERTILE, pal, cmt, maskt, shiftt, cms, masks, shifts);                                   \
+        gDPSetTileSize(pkt, G_TX_RENDERTILE, 0, 0, ((width)-1) << G_TEXTURE_IMAGE_FRAC,                                \
+                       ((height)-1) << G_TEXTURE_IMAGE_FRAC);                                                          \
+    })
```

Comment:
Having trouble implementing this. OoT's implementation is different in that before calling LoadTextureBlock the game happens to look up what `imageSize` to use from an intermediate table, so it hardcodes the index to the `LoadTextureBlock` tables to be the variable `markInfo->imageSize`. 

I can't do this in MM because the size is set directly, and if I try defining the `G_IM_SIZ_MARK` etc. as function like macros the concatenation doesn't work out correctly. I could create multiple macros for all 4 image sizes but that would be far more messy than just having the one macro defined.

Reply:
Yeah I don't think we will be able to avoid the macro. I say keep it but name it something like
`gDPLoadTextureBlockMapIcon` cause it seems to be excusivley used to draw either chest or boss icons.

Reply:
I don't agree with naming it after `MapIcons` or anything map related. The macro has a much more generalized purpose; it's meant to be used in place of `LoadTextureBlock` when `G_IM_SIZ_*b` has been compiled into it's in integer form.

And given how it's used in OoT, the macro and the tables seem like they should actually be split into their own file

Reply:
Does it still match with the file split like that? If it can and the macro could be more generally used (i.e. data is externed and the macro in an h file), then sure a more general name seems appropriate. If not and the data needs to stay in map then it seems better named after map stuff.

Reply:
Could you update this to match your PR over in OoT?

Reply:
Implemented mostly the same as zeldaret/oot#1896 except I stripped out the custom qualifiers option since it isn't necessary